### PR TITLE
add GetBalance by address methods

### PR DESCRIPTION
**Pull request by CodeFunta**

### CODE REVIEW

File: NBXplorer.Client/ExplorerClient.cs [hunk 1]
```diff
@@ -313,7 +313,7 @@ public void CancelReservation(DerivationStrategyBase strategy, KeyPath[] keyPath
 			CancelReservationAsync(strategy, keyPaths, cancellation).GetAwaiter().GetResult();
 		}
 
-		public GetBalanceResponse GetBalance(DerivationStrategyBase userDerivationScheme, CancellationToken cancellation = default)
```
Comment (NicolasDorier) [inline NBXplorer.Client/ExplorerClient.cs:316]: the cancellation token should be passed to `GetBalanceAsync`, not removed.

File: NBXplorer.Client/ExplorerClient.cs [hunk 2]
```diff
@@ -322,6 +322,16 @@ public Task<GetBalanceResponse> GetBalanceAsync(DerivationStrategyBase userDeriv
 			return SendAsync<GetBalanceResponse>(HttpMethod.Get, null, "v1/cryptos/{0}/derivations/{1}/balance", new[] { CryptoCode, userDerivationScheme.ToString() }, cancellation);
 		}
 
+
+		public GetBalanceResponse GetBalance(string address)
```
Comment (NicolasDorier) [inline NBXplorer.Client/ExplorerClient.cs:326]: Should be a strongly typed `BitcoinAddress`, not a string.

File: NBXplorer.Client/ExplorerClient.cs [hunk 3]
```diff
@@ -322,6 +322,16 @@ public Task<GetBalanceResponse> GetBalanceAsync(DerivationStrategyBase userDeriv
...
+		{
+			return GetBalanceAsync(address).GetAwaiter().GetResult();
+		}
+		public Task<GetBalanceResponse> GetBalanceAsync(string address, CancellationToken cancellation = default)
+		{
+			return SendAsync<GetBalanceResponse>(HttpMethod.Get, null, "v1/cryptos/{0}/addresses/{1}/balance", new[] { CryptoCode, address }, cancellation);
```
Comment (NicolasDorier) [inline NBXplorer.Client/ExplorerClient.cs:332]: Reuse `GetBalanceAsync(DerivationStrategyBase userDerivationScheme)` instead, there is a subclass to pass address only.

File: NBXplorer.Client/ExplorerClient.cs (hunk 1 above)
Comment (CodeFunta) [reply NBXplorer.Client/ExplorerClient.cs:316]: Ok, But GetBalance is sync version and cancelationToken don't used inside GetBalanceAsync  call. I'll change it to use

Comment (NicolasDorier) [reply NBXplorer.Client/ExplorerClient.cs:316]: this is a bug, we should use it in `GetBalanceAsync `

File: NBXplorer.Client/ExplorerClient.cs (hunk 2 above)
Comment (CodeFunta) [reply NBXplorer.Client/ExplorerClient.cs:326]: Ok

File: NBXplorer.Client/ExplorerClient.cs (hunk 3 above)
Comment (CodeFunta) [reply NBXplorer.Client/ExplorerClient.cs:332]: I've only seen logic with TrackedSource when address was used. Still don't understand how to create DerivationStrategy from factory for btc address

File: NBXplorer.Client/ExplorerClient.cs (hunk 1 above)
Comment (CodeFunta) [reply NBXplorer.Client/ExplorerClient.cs:316]: fix was pushed

File: NBXplorer.Client/ExplorerClient.cs (hunk 2 above)
Comment (CodeFunta) [reply NBXplorer.Client/ExplorerClient.cs]: Changed with new commit

File: NBXplorer.Client/ExplorerClient.cs (hunk 3 above)
Comment (NicolasDorier) [reply NBXplorer.Client/ExplorerClient.cs]: Ah you are right this is a tracked source, not a derivation strategy.

**Pull request merged**